HW5/server.js

```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const fs = require('fs');
+let app = express();
+app.use(express.static('./files'));
+
+
+app.get('/server/:dir/:fileName',(req,res) => {
+	let dirName = req.params.dir || null;
+	let fileName = req.params.fileName || null;
+	
+	// conditional statements to handle illegal urls
+	if(!dirName){
+		res.send("invalid dir name!");
+	}
+	else if(!fileName){
+		res.send("No file was requested! please enter a file name");
+	}
+	
+	// concat dir name and file name into a single path
+	let path = "./" + dirName + "/" + fileName;
```

Comment:
This is a security fault, read about "path traversal" (just a point to learn from :))

Reply:
so how would you go about making the path, read a bit about path traversal